### PR TITLE
Resolution for issue #642 JDK TCP CONNECTION ISSUE

### PR DESCRIPTION
**Pull request by pmatte1**
Resolution was to use a static reference to the SSL Socket Factory, rather than getting it each time from the SSLContext.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @pmatte1 has signed the CLA. Thanks for the pull request!

**Comment by mattwiller** [comment]
@pmatte1 It looks like there's a ton of reformatting that got done, which makes it really difficult to see what was actually changed here vs. not — could you please update without changing indentation etc from what's on master?  This is also breaking the linter, which causes the CI job to fail.

**Comment by pmatte1** [comment]
@mattwiller let me know if any more needs to be done for this pull request to be reviewed.  Sorry about the formatting.

File: src/main/java/com/box/sdk/BoxAPIRequest.java [hunk 1]
```diff
@@ -58,7 +60,55 @@
     private int numRedirects;
     private boolean followRedirects = true;
     private boolean shouldAuthenticate;
-    private SSLContext sslContext;
+	//private static SSLContext sslContext;
```
Comment (mattwiller) [inline src/main/java/com/box/sdk/BoxAPIRequest.java]: Please remove the commented-out code here

File: src/main/java/com/box/sdk/BoxAPIRequest.java [hunk 2]
```diff
@@ -58,7 +60,55 @@
...
+	private static SSLSocketFactory sslSocketFactory;
```
Comment (mattwiller) [inline src/main/java/com/box/sdk/BoxAPIRequest.java]: This line (and most of the others) are not indented the same as the rest of the file, which is causing the linter to fail.  Before we can merge the PR, all the lint errors will need to be fixed.  See https://travis-ci.org/box/box-java-sdk/jobs/427952040

**Comment by pmatte1** [comment]
@mattwiller Believed I have cleaned up all the indentation issues.  Can you give it a sanity to make sure it all looks OK and the lint errors are all cleaned up?

Thanks in advance.

**Comment by mattwiller** [comment]
@pmatte1 I still see the following lint errors in CI:

```
:checkstyleMain[ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxAPIRequest.java:62:5: Static variable definition in wrong order.
[ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxAPIRequest.java:64:11: '{' is not preceded with whitespace.
[ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxAPIRequest.java:104:11: 'if' is not followed by whitespace.
[ant:checkstyle] /home/travis/build/box/box-java-sdk/src/main/java/com/box/sdk/BoxAPIRequest.java:104:23: '{' is not preceded with whitespace.
```

**Comment by pmatte1** [comment]
@mattwiller Finally figured out how to check the build.  All issues resolved. Ready for review.

**Pull request merged**